src/utils/DependsOnManager.ts

```diff
@@ -36,6 +36,24 @@ export class DependsOnManager {
     $$(this.facet.element).toggleClass('coveo-hidden', !this.parentFacetHasSelectedValues);
   }
 
+  private get dependentFacets() {
+    const allFacets = ComponentsTypes.getAllFacetsInstance(this.facet.root);
+    return allFacets.filter(facet => {
+      return this.facet.id === facet.options.dependsOn;
+    });
+  }
+
+  public get hasDependentFacets() {
+    return !!this.dependentFacets.length;
+  }
+
+  public get dependentFacetsHaveSelectedValues() {
+    return this.dependentFacets.some(facet => {
+      const facetId = QueryStateModel.getFacetId(facet.options.id);
```

Comment:
Not obvious but that's a actually static method building the id correctly for the QSM:
```
static getFacetId(id: string, include: boolean = true) {
    return 'f:' + id + (include ? '' : ':not');
}
```
That line could be moved in the valuesExistForFacetWithId method maybe it would be clearer